Custom view form img

kdaveed opened the pull request:
For #13 and #4

Commit:
There is a new predicate : depictedIn. It appears in the file propertygroup tab, and allows to upload image to the persons. It creates automatically a bibo:Image instance and to it connects the uploaded file with the :hasFile predicate. The edit and delete functions are working as well.

Commit:
Now there is a link for each image instance to its profile page, if it depicts a person. We can add descriptions as well. The bibo:Image profile page is custom, we can upload only one file.

cuboideum commented:
Why is it especially important to upload images of persons?

If I get this right, :depictedIn links persons to images. :isDepicted does the same for all entities. Why is it necessary to have :depictedIn in addition?

What do you mean by "now there is a link for each image instance to its profile page"? What else should there be? Obviously, all links lead to the profile pages of the respective entities.

I do not yet see the point. And I do not quite understand how this relates to issues #13 and #4.

Commit:
Now the <http://www.semanticweb.org/engel/ontologies/2015/3/rdfBones#isDepicted>
is used for uploading images to entities. If an image depicts an entity, then a bibo:Image instance is generated, which we connect the file information containing triple to. If someone wants to upload additional images there, is a link to the profile page of this bibo:Image instance, where arbitrary many new files can be added.

zarquon42b commented:
while this works when images are newly uploaded, I cannot point "is depicted" to an existing image.

Commit:
EditRequestDispatchController.java is in the third tier.

Commit:
The configuration for the custom entry form is done by triples. This commit contains the config of the interface and the corresponding java, javascript and ftl routines. However the custom entry form definition has to contain the triples that are created after the form submission. This comes in the next commit.

Commit:
The configurable custom entry form is ready for the rdfBones:isDepicted predicate. The form offers a possibility to create new instance, as well as assigning existing ones. The edit and delete functions are working as well.

zarquon42b committed:
removed Black,Bog

zarquon42b commented:
@kdaveed:
According to issue #13:
- the custom view form should display a thumbnail next to an image link
- while depicts works, is-depicted only allows file-upload but not linking to an existing image

Please fix this.
